dronegen/container_images.go

```diff
@@ -0,0 +1,942 @@
+				BuiltImage: p.GetStagingRegistryImage(supportedArch, version, stagingRepo),
+				Version:    version,
+				Product:    p,
+			})
+		}
+	}
+
+	for _, containerRepo := range getReposToPublishTo(productionRepos, stagingRepo, flags) {
+		steps = append(steps, containerRepo.buildSteps(archBuildStepDetails)...)
+	}
+
+	return steps
+}
+
+func getReposToPublishTo(productionRepos []*ContainerRepo, stagingRepo *ContainerRepo, flags *TriggerFlags) []*ContainerRepo {
+	stagingRepos := []*ContainerRepo{stagingRepo}
+
+	if flags.ShouldAffectProductionImages {
+		if !flags.ShouldBuildNewImages {
+			// In this case the images will be pulled from staging and therefor should not be re-published
+			// to staging
+			return productionRepos
+		}
+
+		return append(stagingRepos, productionRepos...)
+	}
+
+	return stagingRepos
+}
+
+func (p *Product) GetBuildStepName(arch string, version *releaseVersion) string {
+	telportImageName := p.GetLocalRegistryImage(arch, version)
+	return fmt.Sprintf("Build %s image %q", p.Name, telportImageName.GetDisplayName())
+}
+
+func cleanBuilderName(builderName string) string {
+	var invalidBuildxCharExpression = regexp.MustCompile(`[^a-zA-Z0-9._-]+`)
+	return invalidBuildxCharExpression.ReplaceAllString(builderName, "-")
+}
+
+func (p *Product) createBuildStep(arch string, version *releaseVersion) (step, *buildStepOutput) {
+	localRegistryImage := p.GetLocalRegistryImage(arch, version)
+	builderName := cleanBuilderName(fmt.Sprintf("%s-builder", localRegistryImage.GetDisplayName()))
+
+	buildxConfigFileDir := path.Join("/tmp", builderName)
+	buildxConfigFilePath := path.Join(buildxConfigFileDir, "buildkitd.toml")
+
+	buildxCreateCommand := "docker buildx create"
+	buildxCreateCommand += fmt.Sprintf(" --driver %q", "docker-container")
+	// This is set so that buildx can reach the local registry
+	buildxCreateCommand += fmt.Sprintf(" --driver-opt %q", "network=host")
+	buildxCreateCommand += fmt.Sprintf(" --name %q", builderName)
+	buildxCreateCommand += fmt.Sprintf(" --config %q", buildxConfigFilePath)
+
+	buildCommand := "docker buildx build"
+	buildCommand += " --push"
+	buildCommand += fmt.Sprintf(" --builder %q", builderName)
+	if p.DockerfileTarget != "" {
+		buildCommand += fmt.Sprintf(" --target %q", p.DockerfileTarget)
+	}
+	buildCommand += fmt.Sprintf(" --platform %q", "linux/"+arch)
+	buildCommand += fmt.Sprintf(" --tag %q", localRegistryImage.GetShellName())
+	buildCommand += fmt.Sprintf(" --file %q", p.DockerfilePath)
+	if p.DockerfileArgBuilder != nil {
+		for _, buildArg := range p.DockerfileArgBuilder(arch) {
+			buildCommand += fmt.Sprintf(" --build-arg %q", buildArg)
+		}
+	}
+	buildCommand += " " + p.WorkingDirectory
+
+	step := step{
+		Name:    p.GetBuildStepName(arch, version),
+		Image:   "docker",
+		Volumes: dockerVolumeRefs(),
+		Environment: map[string]value{
+			"DOCKER_BUILDKIT": {
+				raw: "1",
+			},
+		},
+		Commands: []string{
+			"docker run --privileged --rm tonistiigi/binfmt --install all",
```

Comment:
What is this `tonistiigi/binfmt` image?

Reply:
That container ensures `binfmt_misc` support is configured on the host per the `docker buildx` documentation here: https://github.com/docker/buildx#building-multi-platform-images